projects/plugins/jetpack/extensions/blocks/ai-assistant/components/upgrade-prompt/index.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * External dependencies
+ */
+import { isAtomicSite, isSimpleSite } from '@automattic/jetpack-shared-extension-utils';
+import { createInterpolateElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import React from 'react';
+/*
+ * Internal dependencies
+ */
+import { Nudge } from '../../../../shared/components/upgrade-nudge';
+import useAutosaveAndRedirect from '../../../../shared/use-autosave-and-redirect';
+import useUpgradeFlow from '../../../../shared/use-upgrade-flow';
+
+const useAIAssistantUpgradeFlow = () => {
+	// Direct checkout flow for atomic and simple sites, as they do not have My Jetpack.
+	const [ directCheckoutUrl, goToCheckoutPage, isRedirecting ] =
+		useUpgradeFlow( 'jetpack_ai_monthly' );
+
+	// My Jetpack checkout flow for Jetpack sites.
+	const myJetpackCheckoutUrl = `${ window?.Jetpack_Editor_Initial_State?.adminUrl }admin.php?page=my-jetpack#/add-jetpack-ai`;
+	const { autosaveAndRedirect, isRedirecting: isRedirectingMyJetpack } =
+		useAutosaveAndRedirect( myJetpackCheckoutUrl );
```

Comment:
Is it possible it redirects to this URL even for dotcom sites?

Reply:
Nope, the page is `my-jetpack`, it does not exist on dotcom sites. If we register another one somehow, that can be reused just like we do on Jetpack by just changing the URL